File: Samples/Samples/ListView1.cs
using System;
using Xwt;
using Xwt.Drawing;

namespace Samples
{
	public class ListView1: VBox
	{
		DataField<string> name = new DataField<string> ();
		DataField<Image> icon = new DataField<Image> ();
		DataField<string> text = new DataField<string> ();
		DataField<Image> icon2 = new DataField<Image> ();
		
		public ListView1 ()
		{
			PackStart (new Label ("The listview should have a red background"));
			ListView list = new ListView () {
				BackgroundColor = Colors.Red
			};
			ListStore store = new ListStore (name, icon, text, icon2);
			list.DataSource = store;
			list.Columns.Add ("Name", icon, name);
			list.Columns.Add ("Text", icon2, text);
			
			var png = Image.FromResource (typeof(App), "class.png");
			
			for (int n=0; n<100; n++) {
				var r = store.AddRow ();
				store.SetValue (r, icon, png);
				store.SetValue (r, name, "Value " + n);
				store.SetValue (r, icon2, png);
				store.SetValue (r, text, "Text " + n);
			}
			PackStart (list, BoxMode.FillAndExpand);

			list.RowActivated += delegate(object sender, ListViewRowEventArgs e) {
				MessageDialog.ShowMessage ("Row " + e.RowIndex + " activated");
			};
		}
	}
}

File: SamplesVB/Samples/ListView1.vb
Imports System
Imports Xwt
Imports Xwt.Drawing

Namespace Samples
    Public Class ListView1
        Inherits VBox

        Private name As DataField(Of String) = New DataField(Of String)()

        Private icon As DataField(Of Image) = New DataField(Of Image)()

        Private text As DataField(Of String) = New DataField(Of String)()

        Private icon2 As DataField(Of Image) = New DataField(Of Image)()

        Public Sub New()
            Dim list As ListView = New ListView()
            Dim store As ListStore = New ListStore(Me.name, Me.icon, Me.text, Me.icon2)
            list.DataSource = store
            list.Columns.Add("Name", {Me.icon, Me.name})
            list.Columns.Add("Text", {Me.icon2, Me.text})
            Dim png As Image = Image.FromResource(GetType(App), "class.png")
            For i As Integer = 0 To 100 - 1
                Dim r As Integer = store.AddRow()
                store.SetValue(Of Image)(r, Me.icon, png)
                store.SetValue(Of String)(r, Me.name, "Value " + i)
                store.SetValue(Of Image)(r, Me.icon2, png)
                store.SetValue(Of String)(r, Me.text, "Text " + i)
            Next
            MyBase.PackStart(list, BoxMode.FillAndExpand)
        End Sub
    End Class
End Namespace
